h5pt.go

```diff
@@ -143,91 +171,105 @@ func extractValues(ad *appendData, data interface{}) error {
 		ad.numRecords = 1
 		stringData := C.CString(rv.String())
 		ad.strs = append(ad.strs, unsafe.Pointer(stringData))
-		dataPtr = unsafe.Pointer(&stringData)
-		dataSize = C.size_t(unsafe.Sizeof(dataPtr))
+		dataPtr := uintptr(unsafe.Pointer(&stringData))
```

Comment:
I am wondering whether we couldn't get away with:

```go
str := append([]byte(rv.String()), 0)
ad.strs = append(ad.strs, unsafe.Pointer(&str[0]))
dataPtr := uintptr(ad.strs[len(ad.strs)-1])
```

then we don't have to worry about `C.free` anymore...

Reply:
It seems to work.
But we do not need to append it to the ad.strs any longer. Patch incoming!